### PR TITLE
CP-14154: Check feature-suspend for some operations

### PR DESCRIPTION
**Pull request by thomassa**
For operations
suspend, checkpoint, pool_migrate, and migrate_send
on running HVM guests with no valid PV drivers version,
we now require the "feature-suspend" flag from the guest.

This reverses the spirit of commit
55c412107bb82b06c1158798096e75168a659d45

### CODE REVIEW

File: ocaml/xapi/xapi_vm_lifecycle.ml [hunk 1]
```diff
@@ -168,6 +169,9 @@ let check_op_for_feature ~__context ~vmr ~vmgmr ~power_state ~op ~ref ~strict =
 			| `changing_VCPUs_live
 					when lack_feature "feature-vcpu-hotplug"
 						-> some_err Api_errors.vm_lacks_feature_vcpu_hotplug
+			| `suspend | `checkpoint | `pool_migrate | `migrate_send
```
Comment (thomassa) [inline ocaml/xapi/xapi_vm_lifecycle.ml]: Do you think it's worth a comment to make explicit that (in accordance with the OCaml language specification) the `when` guard applies to all four preceding operations, not just the last one?

**Comment by thomassa** [comment]
I'm putting this here for review, but please don't merge it just yet: I'm spinning up a set-up where I'll check that the new behaviour is as expected.

File: ocaml/xapi/xapi_vm_lifecycle.ml (hunk 1 above)
Comment (robhoes) [inline ocaml/xapi/xapi_vm_lifecycle.ml]: I didn't know that :)

Comment (johnelse) [inline ocaml/xapi/xapi_vm_lifecycle.ml]: Neither did I :) I checked it in utop though, and @thomassa is right!

Comment (thomassa) [inline ocaml/xapi/xapi_vm_lifecycle.ml]: Yes, I checked that it worked, and also looked it up on Inria to check that it was officially specified: I wouldn't have wanted to rely on it if it had been unspecified behaviour that just happened to work in the current OCaml implementation.

**Comment by robhoes** [comment]
Patch looks good, so I'll merge it now.

**Pull request merged**